plugins/USBqemu/qemu-usb/usb-ohci.cpp

```diff
@@ -356,7 +356,7 @@ static void ohci_copy_td(OHCIState *ohci, struct ohci_td *td,
 
     ptr = td->cbp;
     n = 0x1000 - (ptr & 0xfff);
-    if (n > len)
+    if (n > static_cast<uint32_t>(len))
```

Comment:
> Can len be negative ? 

Nope, not possible. it is only called twice : 

**At first case**,
``ohci_copy_td(ohci, &td, ohci->usb_buf, len, 0);``

``len`` uses a type of ``size_t`` (AKA ``unsigned int``), so no possibilities for negative values.

**At second case**,
  ```
if (ret >= 0) {
        if (dir == OHCI_TD_DIR_IN) {
            ohci_copy_td(ohci, &td, ohci->usb_buf, ret, 1);
#ifdef DEBUG_PACKET
            dprintf("  data:");
            for (i = 0; i < ret; i++)
                printf(" %.2x", ohci->usb_buf[i]);
            dprintf("\n");
#endif
        } else {
            ret = len;
        }
    }

```
once again, not possible as the function will only get called if ``ret`` is a positive value.

> Maybe it could be better to change the parameter of the function instead ? 

I didn't care much about the USB code so I left it just like that also partially due to laziness, should I change the parameter of the function ? (your choice)